pennylane/qnn/keras.py

```diff
@@ -198,7 +199,32 @@ def __init__(
                 "https://www.tensorflow.org/install for detailed instructions."
             )
 
-        self.sig = qnode.func.sig
+        self.weight_shapes = {
+            weight: (tuple(size) if isinstance(size, Iterable) else (size,) if size > 1 else ())
+            for weight, size in weight_shapes.items()
+        }
+
+        if qml.tape_mode_active():
+            self._signature_validation_tape_mode(qnode, weight_shapes)
+            self.qnode = qnode
+
+            dtype = tf.float32 if tf.keras.backend.floatx() == tf.float32 else tf.float64
+            self.qnode.to_tf(dtype=dtype)
```

Comment:
I have no idea why, but `self.qnode.to_tf(dtype=tf.keras.backend.floatx())` causes an error when calculating the gradient. The error looks like we're setting the `dtype` to be `tf.float64` when it's expecting `tf.float32`. However, I print out `tf.keras.backend.floatx()` and it always is `tf.float32` :shrug: